src/masternode-budget.cpp

```diff
@@ -234,10 +230,9 @@ void CBudgetManager::SubmitFinalBudget()
         return;
     }
 
-    LOCK(cs);
     AddSeenFinalizedBudget(finalizedBudgetBroadcast);
     finalizedBudgetBroadcast.Relay();
-    budget.AddFinalizedBudget(finalizedBudgetBroadcast);
+    AddFinalizedBudget(finalizedBudgetBroadcast);
```

Comment:
line 167 has one more instance calling to `budget.GetBudget()` that can be cleaned as well.

Reply:
Yep, this too is fixed in some commit down the road.
By #1861 there won't be any direct access to the global budget manager from within budget-related objects/classes.